pandas-stubs/io/formats/style.pyi

```diff
@@ -1,97 +1,333 @@
+        hrules: bool | None = ...,
+        clines: Literal["all;data", "all;index", "skip-last;data", "skip-last;index"]
+        | None = ...,
+        label: str | None = ...,
+        caption: str | tuple[str, str] | None = ...,
+        sparse_index: bool | None = ...,
+        sparse_columns: bool | None = ...,
+        multirow_align: Literal["c", "t", "b", "naive"] | None = ...,
+        multicol_align: Literal["r", "c", "l", "naive-l", "naive-r"] | None = ...,
+        siunitx: bool = ...,
+        environment: str | None = ...,
+        encoding: str | None = ...,
+        convert_css: bool = ...,
+    ) -> str: ...
+    @overload
+    def to_html(
+        self,
+        buf: FilePath | WriteBuffer[str],
+        *,
+        table_uuid: str | None = ...,
+        table_attributes: str | None = ...,
+        sparse_index: bool | None = ...,
+        sparse_columns: bool | None = ...,
+        bold_headers: bool = ...,
+        caption: str | None = ...,
+        max_rows: int | None = ...,
+        max_columns: int | None = ...,
+        encoding: str | None = ...,
+        doctype_html: bool = ...,
+        exclude_styles: bool = ...,
+        **kwargs: Any,
+    ) -> None: ...
+    @overload
+    def to_html(
+        self,
+        buf: None = ...,
+        *,
+        table_uuid: str | None = ...,
+        table_attributes: str | None = ...,
+        sparse_index: bool | None = ...,
+        sparse_columns: bool | None = ...,
+        bold_headers: bool = ...,
+        caption: str | None = ...,
+        max_rows: int | None = ...,
+        max_columns: int | None = ...,
+        encoding: str | None = ...,
+        doctype_html: bool = ...,
+        exclude_styles: bool = ...,
+        **kwargs: Any,
+    ) -> str: ...
+    def set_td_classes(self, classes: DataFrame) -> Styler: ...
+    def __copy__(self) -> Styler: ...
+    def __deepcopy__(self, memo) -> Styler: ...
     def clear(self) -> None: ...
-    def apply(self, func, axis: int = ..., subset=..., **kwargs): ...
-    def applymap(self, func, subset=..., **kwargs): ...
-    def where(self, cond, value, other=..., subset=..., **kwargs): ...
-    def set_precision(self, precision): ...
-    def set_table_attributes(self, attributes): ...
-    def export(self): ...
-    def use(self, styles): ...
-    def set_uuid(self, uuid): ...
-    def set_caption(self, caption): ...
-    def set_table_styles(self, table_styles): ...
-    def set_na_rep(self, na_rep: str) -> Styler: ...
-    def hide_index(self): ...
-    def hide_columns(self, subset): ...
-    def highlight_null(self, null_color: str = ...): ...
+    @overload
+    def apply(
+        self,
+        func: Callable[[Series], list | Series],
+        axis: Axis = ...,
+        subset: Subset | None = ...,
+        **kwargs: Any,
+    ) -> Styler: ...
+    @overload
+    def apply(
+        self,
+        func: Callable[[DataFrame], npt.NDArray | DataFrame],
```

Comment:
Wouldn't the following avoid the `overload` ?
```python
       func: Callable[[Series], list | Series] | Callable[[DataFrame], npt.NDArray | DataFrame]
```

Reply:
The overload here is a case of adding specificity. The `axis` determines which type of function is required. Note that is is `axis : Axis=...` (default) in one, and `axis : None` in the other.